airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresDebugger.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.postgres;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.cdk.integrations.debug.AbstractSourceDebugger;
+
+public class PostgresDebugger extends AbstractSourceDebugger {
+
+  @SuppressWarnings({"unchecked", "deprecation", "resource"})
+  public static void main(final String[] args) throws Exception {
+    final PostgresDebugger debugger = new PostgresDebugger();
+    debugger.check();
+    debugger.discover();
+    debugger.read();
+  }
+
+  PostgresDebugger() throws Exception {
+    super();
+  }
+
+  @Override
+  protected PostgresSource getSource() {
+    return new PostgresSource();
+  }
+
+  @Override
+  protected boolean perStreamEnabled() {
+    return true;
+  }
+
+  @Override
+  protected JsonNode convertToDebugConfig(final JsonNode originalConfig) {
+    if (!PostgresUtils.shouldFlushAfterSync(originalConfig)) {
+      throw new RuntimeException("WARNING: config indicates that we are clearing the WAL log while reading data. This will mutate the WAL log"
+          + " associated with the source being debugged and is not advised.");
+    }
```

Comment:
Good point. Not as `check()` since the connector protocol methods are meant to run in isolation. The actual mutation is being done in `read()` so I've moved the logic there